network/p2p/gossip/bloom.go

```diff
@@ -47,49 +58,50 @@ type BloomFilter struct {
 
 func (b *BloomFilter) Add(gossipable Gossipable) {
 	h := gossipable.GossipID()
-	salted := &hasher{
-		hash: h[:],
-		salt: b.salt,
-	}
-	b.bloom.Add(salted)
+	bloom.Add(b.bloom, h[:], b.salt[:])
 }
 
 func (b *BloomFilter) Has(gossipable Gossipable) bool {
 	h := gossipable.GossipID()
-	salted := &hasher{
-		hash: h[:],
-		salt: b.salt,
-	}
-	return b.bloom.Contains(salted)
+	return bloom.Contains(b.bloom, h[:], b.salt[:])
 }
 
+// TODO: Remove error from the return
 func (b *BloomFilter) Marshal() ([]byte, []byte, error) {
-	bloomBytes, err := b.bloom.MarshalBinary()
+	bloomBytes := b.bloom.Marshal()
 	// salt must be copied here to ensure the bytes aren't overwritten if salt
 	// is later modified.
 	salt := b.salt
-	return bloomBytes, salt[:], err
+	return bloomBytes, salt[:], nil
 }
 
-// ResetBloomFilterIfNeeded resets a bloom filter if it breaches a target false
-// positive probability. Returns true if the bloom filter was reset.
+// ResetBloomFilterIfNeeded resets a bloom filter if it breaches [targetFalsePositiveProbability].
+//
+// If [targetElements] exceeds [minTargetElements], the size of the bloom filter will grow to maintain
+// the same [targetFalsePositiveProbability].
+//
+// Returns true if the bloom filter was reset.
 func ResetBloomFilterIfNeeded(
 	bloomFilter *BloomFilter,
-	falsePositiveProbability float64,
+	targetElements int,
 ) (bool, error) {
-	if bloomFilter.bloom.FalsePosititveProbability() < falsePositiveProbability {
+	if bloomFilter.bloom.Count() < bloomFilter.maxCount {
```

Comment:
Super nitty - but I fell like this should be
```suggestion
	if bloomFilter.bloom.Count() <= bloomFilter.maxCount {
```